cmd/compute-domain-kubelet-plugin/driver.go

```diff
@@ -171,73 +173,75 @@ func (d *driver) NodeUnprepareResources(ctx context.Context, req *drapbv1.NodeUn
 
 	workQueue.Run(ctx)
 
-	return unpreparedResources, nil
+	return results, nil
 }
 
-func (d *driver) nodePrepareResource(ctx context.Context, claim *drapbv1.Claim) (bool, *drapbv1.NodePrepareResourceResponse) {
+func (d *driver) nodePrepareResource(ctx context.Context, claim *resourceapi.ResourceClaim) (bool, kubeletplugin.PrepareResult) {
 	d.Lock()
 	defer d.Unlock()
 
-	resourceClaim, err := d.client.ResourceV1beta1().ResourceClaims(claim.Namespace).Get(
-		ctx,
-		claim.Name,
-		metav1.GetOptions{})
-	if err != nil {
-		ret := &drapbv1.NodePrepareResourceResponse{
-			Error: fmt.Sprintf("failed to fetch ResourceClaim %s in namespace %s", claim.Name, claim.Namespace),
-		}
-		return isPermanentError(err), ret
-	}
-
-	if resourceClaim.Status.Allocation == nil {
-		ret := &drapbv1.NodePrepareResourceResponse{
-			Error: fmt.Sprintf("no allocation set in ResourceClaim %s in namespace %s", claim.Name, claim.Namespace),
+	if claim.Status.Allocation == nil {
+		res := kubeletplugin.PrepareResult{
+			Err: fmt.Errorf("no allocation set in ResourceClaim %s in namespace %s", claim.Name, claim.Namespace),
 		}
-		return true, ret
+		return true, res
 	}
 
-	prepared, err := d.state.Prepare(ctx, resourceClaim)
+	devs, err := d.state.Prepare(ctx, claim)
 	if err != nil {
-		ret := &drapbv1.NodePrepareResourceResponse{
-			Error: fmt.Sprintf("error preparing devices for claim %v: %v", claim.UID, err),
+		res := kubeletplugin.PrepareResult{
+			Err: fmt.Errorf("error preparing devices for claim %v: %v", claim.UID, err),
 		}
-		return isPermanentError(err), ret
+		return isPermanentError(err), res
+	}
+
+	// Translate type: drapbv1.Device -> kubeletplugin.Device
+	// Maybe instead change return type of state.Prepare()
+	var prepDevs []kubeletplugin.Device
+	for _, d := range devs {
+		device := kubeletplugin.Device{
+			// kubeletplugin/draplugin.go has a stripSubrequestNames() helper:
+			// Requests: stripSubrequestNames(d.RequestNames),
+			Requests:     d.RequestNames,
```

Comment:
@klueska do we have to filter/strip `d.RequestNames` here?

Reply:
note to self: I would still love to clarify this before merge

Reply:
update: IIUC we do not need to look into stripping subrequest names in our driver code as of today.

Seemingly the context is https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/4816-dra-prioritized-list/README.md.

Reply:
>  all direct manipulation / caching of the type drapbv1.Claim from the driver code itself

here we meant `drapbv1.Device`